docs/BROKER_SPEC.md

```diff
@@ -0,0 +1,109 @@
+# Broker Specification (v0.1.0)
```

Comment:
Would middleware be another name option? It reminds me a lot of middleware in web application frameworks

Reply:
Yeah, you are right, middleware is a good name